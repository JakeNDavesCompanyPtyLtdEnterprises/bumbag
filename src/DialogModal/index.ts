export { default } from './DialogModal';
export { default as DialogModal } from './DialogModal';
