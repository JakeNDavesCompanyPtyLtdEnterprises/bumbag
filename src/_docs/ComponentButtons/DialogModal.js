import React from 'react';

const SvgComponent = props => (
  <svg width={105} height={60} fill="none" {...props}>
    <path fill="#3926A5" d="M0 0h105v60H0z" />
    <g clipPath="url(#prefix__clip0)">
      <path fill="#fff" d="M6 5h92.308v50H6z" />
      <path d="M6.385 19.583h91.538M6.385 42.083h91.538" stroke="#08175D" strokeLinecap="square" />
      <path
        d="M94.846 5.417H9.462c-1.7 0-3.077 1.492-3.077 3.333v42.5c0 1.841 1.377 3.333 3.077 3.333h85.384c1.7 0 3.077-1.492 3.077-3.333V8.75c0-1.84-1.377-3.333-3.077-3.333z"
        stroke="#08175D"
      />
      <path
        d="M73.126 12H15v2.807h58.126V12zM90.564 24.855H15v3.65h75.564v-3.65zM52.782 32.518H15v3.649h37.782v-3.65z"
        fill="#3926A5"
        fillOpacity={0.12}
      />
      <path
        d="M89.314 44.23H73.692c-.85 0-1.538.747-1.538 1.667v4.584c0 .92.689 1.666 1.538 1.666h15.622c.85 0 1.538-.746 1.538-1.666v-4.584c0-.92-.689-1.666-1.538-1.666z"
        fill="#3926A5"
        stroke="#08175D"
        strokeWidth={0.5}
      />
      <path d="M88.973 47.699H74.18v1.225h14.793V47.7z" fill="#fff" />
      <path
        d="M67.923 44.23h-15.77c-.849 0-1.538.747-1.538 1.667v4.584c0 .92.69 1.666 1.539 1.666h15.77c.849 0 1.537-.746 1.537-1.666v-4.584c0-.92-.688-1.666-1.538-1.666z"
        fill="#3926A5"
        fillOpacity={0.12}
        stroke="#08175D"
        strokeWidth={0.5}
      />
      <path d="M67.567 47.699H52.659v1.225h14.908V47.7z" fill="#3926A5" />
    </g>
    <defs>
      <clipPath id="prefix__clip0">
        <path fill="#fff" transform="translate(6 5)" d="M0 0h92.308v50H0z" />
      </clipPath>
    </defs>
  </svg>
);

export default SvgComponent;
