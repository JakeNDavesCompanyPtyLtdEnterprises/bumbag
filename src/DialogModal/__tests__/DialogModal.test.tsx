import * as React from 'react';
import render from '../../_utils/tests/render';
import Modal from '../../Modal';
import DialogModal from '../DialogModal';

jest.mock('reakit/Portal', () => 'portal');

it('renders correctly for a default dialog modal', () => {
  const { container } = render(
    <Modal.Container defaultVisible>
      {modal => <DialogModal {...modal}>This is the content</DialogModal>}
    </Modal.Container>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a dialog modal with a close button', () => {
  const { container } = render(
    <Modal.Container defaultVisible>
      {modal => (
        <DialogModal title="This is a title" showCloseButton {...modal}>
          This is the content
        </DialogModal>
      )}
    </Modal.Container>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a dialog modal with a title', () => {
  const { container } = render(
    <Modal.Container defaultVisible>
      {modal => (
        <DialogModal title="This is a title" {...modal}>
          This is the content
        </DialogModal>
      )}
    </Modal.Container>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a dialog modal with action buttons', () => {
  const { container } = render(
    <Modal.Container defaultVisible>
      {modal => (
        <DialogModal title="This is a title" showActionButtons {...modal}>
          This is the content
        </DialogModal>
      )}
    </Modal.Container>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a dialog modal with a footer', () => {
  const { container } = render(
    <Modal.Container defaultVisible>
      {modal => (
        <DialogModal footer="This is a footer" title="This is a title" {...modal}>
          This is the content
        </DialogModal>
      )}
    </Modal.Container>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a dialog modal with kind alert', () => {
  const { container } = render(
    <Modal.Container defaultVisible>
      {modal => (
        <DialogModal kind="alert" {...modal}>
          This is the content
        </DialogModal>
      )}
    </Modal.Container>
  );
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for a dialog modal with type', () => {
  const { container } = render(
    <Modal.Container defaultVisible>
      {modal => (
        <DialogModal type="danger" {...modal}>
          This is the content
        </DialogModal>
      )}
    </Modal.Container>
  );
  expect(container.firstChild).toMatchSnapshot();
});
