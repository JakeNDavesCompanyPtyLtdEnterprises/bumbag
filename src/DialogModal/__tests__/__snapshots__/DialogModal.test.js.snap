// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for a default dialog modal 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 998;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  -moz-tap-highlight-color: rgba(0,0,0,0);
  background-color: black;
  opacity: 0.4;
}

.c0[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

.c3 {
  -webkit-transform: translate3d(-50%,-50%,0px);
  -ms-transform: translate3d(-50%,-50%,0px);
  transform: translate3d(-50%,-50%,0px);
  position: fixed;
  z-index: 19900410;
  left: 50%;
  top: 50%;
}

.c3[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

.c2 {
  padding: 0 1rem;
  max-width: 600px;
  width: 100%;
}

.c5 {
  background-color: white;
  border-radius: 3px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border: 1px solid #e6e6e6;
}

.c6 {
  padding: 1.25rem;
  max-height: 50vh;
  overflow-y: scroll;
  width: 100%;
}

.c4 {
  border-radius: 5px;
}

.c4:focus {
  outline: unset;
}

<portal>
  <div>
    <button
      aria-hidden="false"
      class="c0 c1 "
      role="button"
      tabindex="-1"
    />
    <div
      aria-hidden="false"
      aria-modal="true"
      class="c2 c3 c1"
      role="dialog"
    >
      <div
        aria-describedby="Modal-3"
        aria-labelledby="Modal-4"
        class="c4 c5 c1"
        role="dialog"
        style="width: 100%;"
        tabindex="-1"
        width="100%"
      >
        <div
          class="c6 c1"
          id="Modal-3"
        >
          This is the content
        </div>
      </div>
    </div>
  </div>
</portal>
`;

exports[`renders correctly for a dialog modal with a close button 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c11 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c0 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 998;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  -moz-tap-highlight-color: rgba(0,0,0,0);
  background-color: black;
  opacity: 0.4;
}

.c0[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

.c3 {
  -webkit-transform: translate3d(-50%,-50%,0px);
  -ms-transform: translate3d(-50%,-50%,0px);
  transform: translate3d(-50%,-50%,0px);
  position: fixed;
  z-index: 19900410;
  left: 50%;
  top: 50%;
}

.c3[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

.c2 {
  padding: 0 1rem;
  max-width: 600px;
  width: 100%;
}

.c10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border: 1px solid #bfbfbf;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c10:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #7467c0 0px 0px 0px 2px;
}

.c10[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c10 {
  border: 0;
  background: unset;
  color: #435a6f;
  fill: #435a6f;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c10:hover {
  color: #000;
  fill: #000;
}

.c14 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

.c8 {
  font-weight: 700;
  line-height: 1.2;
  margin-bottom: 1rem;
}

.heading.c8 + .sub-heading {
  margin-top: -1rem;
}

.c8:last-child {
  margin-bottom: 0;
}

h1.c8 {
  font-size: 3rem;
}

h2.c8 {
  font-size: 2.5rem;
}

h3.c8 {
  font-size: 2rem;
}

h4.c8 {
  font-size: 1.5rem;
}

h5.c8 {
  font-size: 1.25rem;
}

h6.c8 {
  font-size: 1rem;
}

.c8 .c13 {
  top: 0.1em;
}

.c8 {
  font-weight: 600;
}

.c5 {
  background-color: white;
  border-radius: 3px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border: 1px solid #e6e6e6;
}

.c15 {
  padding: 1.25rem;
  max-height: 50vh;
  overflow-y: scroll;
  width: 100%;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid #e6e6e6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 1.25rem;
  width: 100%;
}

.c7 {
  margin-bottom: 0px;
}

.c9 {
  height: 2em;
  width: 2em;
  padding: 0px;
}

.c9:hover {
  background-color: #f2f2f2;
}

.c4 {
  border-radius: 5px;
}

.c4:focus {
  outline: unset;
}

.c12 {
  border: 0px;
  -webkit-clip: rect(0px,0px,0px,0px);
  clip: rect(0px,0px,0px,0px);
  height: 1px;
  width: 1px;
  margin: -1px;
  padding: 0px;
  overflow: hidden;
  position: absolute;
}

@media (max-width:480px) {
  h1.c8 {
    font-size: 2rem;
  }
}

@media (max-width:480px) {
  h2.c8 {
    font-size: 1.8rem;
  }
}

@media (max-width:480px) {
  h3.c8 {
    font-size: 1.6rem;
  }
}

@media (max-width:480px) {
  h4.c8 {
    font-size: 1.4rem;
  }
}

@media (max-width:480px) {
  h5.c8 {
    font-size: 1.2rem;
  }
}

<portal>
  <div>
    <button
      aria-hidden="false"
      class="c0 c1 "
      role="button"
      tabindex="-1"
    />
    <div
      aria-hidden="false"
      aria-modal="true"
      class="c2 c3 c1"
      role="dialog"
    >
      <div
        aria-describedby="Modal-3"
        aria-labelledby="Modal-4"
        class="c4 c5 c1"
        role="dialog"
        style="width: 100%;"
        tabindex="-1"
        width="100%"
      >
        <div
          class="c6 c1"
        >
          <h5
            class="c7 sub-heading c8 c1"
            id="Modal-4"
          >
            This is a title
          </h5>
          <button
            class="c9 c10 c11"
            kind="link"
            style="size: default;"
            type="button"
          >
            <div
              class="c12 c1"
            >
              Close
            </div>
            <svg
              aria-hidden="true"
              class="c13 c14 c1"
              role="img"
              size="1"
              style="size: 1px;"
              viewBox="0 0 16 16"
            >
              <path
                d="M9.41 8l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 0 0-1.71-.71L8 6.59l-3.29-3.3a1.003 1.003 0 0 0-1.42 1.42L6.59 8 3.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 0 0 1.71.71L8 9.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 0 0 .71-1.71L9.41 8z"
                fill-rule="even-odd"
              />
            </svg>
          </button>
        </div>
        <div
          class="c15 c1"
          id="Modal-3"
        >
          This is the content
        </div>
      </div>
    </div>
  </div>
</portal>
`;

exports[`renders correctly for a dialog modal with a footer 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 998;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  -moz-tap-highlight-color: rgba(0,0,0,0);
  background-color: black;
  opacity: 0.4;
}

.c0[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

.c3 {
  -webkit-transform: translate3d(-50%,-50%,0px);
  -ms-transform: translate3d(-50%,-50%,0px);
  transform: translate3d(-50%,-50%,0px);
  position: fixed;
  z-index: 19900410;
  left: 50%;
  top: 50%;
}

.c3[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

.c2 {
  padding: 0 1rem;
  max-width: 600px;
  width: 100%;
}

.c8 {
  font-weight: 700;
  line-height: 1.2;
  margin-bottom: 1rem;
}

.heading.c8 + .sub-heading {
  margin-top: -1rem;
}

.c8:last-child {
  margin-bottom: 0;
}

h1.c8 {
  font-size: 3rem;
}

h2.c8 {
  font-size: 2.5rem;
}

h3.c8 {
  font-size: 2rem;
}

h4.c8 {
  font-size: 1.5rem;
}

h5.c8 {
  font-size: 1.25rem;
}

h6.c8 {
  font-size: 1rem;
}

.c8 .c11 {
  top: 0.1em;
}

.c8 {
  font-weight: 600;
}

.c5 {
  background-color: white;
  border-radius: 3px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border: 1px solid #e6e6e6;
}

.c9 {
  padding: 1.25rem;
  max-height: 50vh;
  overflow-y: scroll;
  width: 100%;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid #e6e6e6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 1.25rem;
  width: 100%;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-top: 1px solid #e6e6e6;
  padding: 1rem 1rem;
  width: 100%;
}

.c7 {
  margin-bottom: 0px;
}

.c4 {
  border-radius: 5px;
}

.c4:focus {
  outline: unset;
}

@media (max-width:480px) {
  h1.c8 {
    font-size: 2rem;
  }
}

@media (max-width:480px) {
  h2.c8 {
    font-size: 1.8rem;
  }
}

@media (max-width:480px) {
  h3.c8 {
    font-size: 1.6rem;
  }
}

@media (max-width:480px) {
  h4.c8 {
    font-size: 1.4rem;
  }
}

@media (max-width:480px) {
  h5.c8 {
    font-size: 1.2rem;
  }
}

<portal>
  <div>
    <button
      aria-hidden="false"
      class="c0 c1 "
      role="button"
      tabindex="-1"
    />
    <div
      aria-hidden="false"
      aria-modal="true"
      class="c2 c3 c1"
      role="dialog"
    >
      <div
        aria-describedby="Modal-3"
        aria-labelledby="Modal-4"
        class="c4 c5 c1"
        role="dialog"
        style="width: 100%;"
        tabindex="-1"
        width="100%"
      >
        <div
          class="c6 c1"
        >
          <h5
            class="c7 sub-heading c8 c1"
            id="Modal-4"
          >
            This is a title
          </h5>
        </div>
        <div
          class="c9 c1"
          id="Modal-3"
        >
          This is the content
        </div>
        <div
          class="c10 c1"
          style="justify-content: space-between;"
        >
          <div
            class="c1"
          >
            This is a footer
          </div>
        </div>
      </div>
    </div>
  </div>
</portal>
`;

exports[`renders correctly for a dialog modal with a title 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 998;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  -moz-tap-highlight-color: rgba(0,0,0,0);
  background-color: black;
  opacity: 0.4;
}

.c0[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

.c3 {
  -webkit-transform: translate3d(-50%,-50%,0px);
  -ms-transform: translate3d(-50%,-50%,0px);
  transform: translate3d(-50%,-50%,0px);
  position: fixed;
  z-index: 19900410;
  left: 50%;
  top: 50%;
}

.c3[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

.c2 {
  padding: 0 1rem;
  max-width: 600px;
  width: 100%;
}

.c8 {
  font-weight: 700;
  line-height: 1.2;
  margin-bottom: 1rem;
}

.heading.c8 + .sub-heading {
  margin-top: -1rem;
}

.c8:last-child {
  margin-bottom: 0;
}

h1.c8 {
  font-size: 3rem;
}

h2.c8 {
  font-size: 2.5rem;
}

h3.c8 {
  font-size: 2rem;
}

h4.c8 {
  font-size: 1.5rem;
}

h5.c8 {
  font-size: 1.25rem;
}

h6.c8 {
  font-size: 1rem;
}

.c8 .c10 {
  top: 0.1em;
}

.c8 {
  font-weight: 600;
}

.c5 {
  background-color: white;
  border-radius: 3px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border: 1px solid #e6e6e6;
}

.c9 {
  padding: 1.25rem;
  max-height: 50vh;
  overflow-y: scroll;
  width: 100%;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid #e6e6e6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 1.25rem;
  width: 100%;
}

.c7 {
  margin-bottom: 0px;
}

.c4 {
  border-radius: 5px;
}

.c4:focus {
  outline: unset;
}

@media (max-width:480px) {
  h1.c8 {
    font-size: 2rem;
  }
}

@media (max-width:480px) {
  h2.c8 {
    font-size: 1.8rem;
  }
}

@media (max-width:480px) {
  h3.c8 {
    font-size: 1.6rem;
  }
}

@media (max-width:480px) {
  h4.c8 {
    font-size: 1.4rem;
  }
}

@media (max-width:480px) {
  h5.c8 {
    font-size: 1.2rem;
  }
}

<portal>
  <div>
    <button
      aria-hidden="false"
      class="c0 c1 "
      role="button"
      tabindex="-1"
    />
    <div
      aria-hidden="false"
      aria-modal="true"
      class="c2 c3 c1"
      role="dialog"
    >
      <div
        aria-describedby="Modal-3"
        aria-labelledby="Modal-4"
        class="c4 c5 c1"
        role="dialog"
        style="width: 100%;"
        tabindex="-1"
        width="100%"
      >
        <div
          class="c6 c1"
        >
          <h5
            class="c7 sub-heading c8 c1"
            id="Modal-4"
          >
            This is a title
          </h5>
        </div>
        <div
          class="c9 c1"
          id="Modal-3"
        >
          This is the content
        </div>
      </div>
    </div>
  </div>
</portal>
`;

exports[`renders correctly for a dialog modal with action buttons 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c13 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #fcfcfc;
  color: inherit;
}

.c15 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: #3926a5;
  color: inherit;
}

.c0 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 998;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  -moz-tap-highlight-color: rgba(0,0,0,0);
  background-color: black;
  opacity: 0.4;
}

.c0[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

.c3 {
  -webkit-transform: translate3d(-50%,-50%,0px);
  -ms-transform: translate3d(-50%,-50%,0px);
  transform: translate3d(-50%,-50%,0px);
  position: fixed;
  z-index: 19900410;
  left: 50%;
  top: 50%;
}

.c3[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

.c2 {
  padding: 0 1rem;
  max-width: 600px;
  width: 100%;
}

.c12 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fcfcfc;
  border: 1px solid #bfbfbf;
  border-radius: 4px;
  color: #435a6f;
  fill: #435a6f;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c12:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #7467c0 0px 0px 0px 2px;
}

.c12[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c12:hover {
  background-color: #efefef;
}

.c12:hover:active {
  background-color: #e3e3e3;
}

.c14 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #3926a5;
  border: 1px solid #1c1352;
  border-radius: 4px;
  color: #fff;
  fill: #fff;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-weight: 600;
  height: 2.5em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c14:focus {
  outline: unset;
  z-index: 2;
  box-shadow: #7467c0 0px 0px 0px 2px;
}

.c14[disabled] {
  cursor: not-allowed;
  opacity: 0.7;
  outline: unset;
  pointer-events: unset;
}

.c14:hover {
  background-color: #322190;
}

.c14:hover:active {
  background-color: #2b1c7c;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-left: -0.5rem;
  margin-top: -0.5rem;
}

.c11 > * {
  margin-left: 0.5rem;
  margin-top: 0.5rem;
}

.c8 {
  font-weight: 700;
  line-height: 1.2;
  margin-bottom: 1rem;
}

.heading.c8 + .sub-heading {
  margin-top: -1rem;
}

.c8:last-child {
  margin-bottom: 0;
}

h1.c8 {
  font-size: 3rem;
}

h2.c8 {
  font-size: 2.5rem;
}

h3.c8 {
  font-size: 2rem;
}

h4.c8 {
  font-size: 1.5rem;
}

h5.c8 {
  font-size: 1.25rem;
}

h6.c8 {
  font-size: 1rem;
}

.c8 .c16 {
  top: 0.1em;
}

.c8 {
  font-weight: 600;
}

.c5 {
  background-color: white;
  border-radius: 3px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border: 1px solid #e6e6e6;
}

.c9 {
  padding: 1.25rem;
  max-height: 50vh;
  overflow-y: scroll;
  width: 100%;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid #e6e6e6;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 1.25rem;
  width: 100%;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-top: 1px solid #e6e6e6;
  padding: 1rem 1rem;
  width: 100%;
}

.c7 {
  margin-bottom: 0px;
}

.c4 {
  border-radius: 5px;
}

.c4:focus {
  outline: unset;
}

@media (max-width:480px) {
  h1.c8 {
    font-size: 2rem;
  }
}

@media (max-width:480px) {
  h2.c8 {
    font-size: 1.8rem;
  }
}

@media (max-width:480px) {
  h3.c8 {
    font-size: 1.6rem;
  }
}

@media (max-width:480px) {
  h4.c8 {
    font-size: 1.4rem;
  }
}

@media (max-width:480px) {
  h5.c8 {
    font-size: 1.2rem;
  }
}

<portal>
  <div>
    <button
      aria-hidden="false"
      class="c0 c1 "
      role="button"
      tabindex="-1"
    />
    <div
      aria-hidden="false"
      aria-modal="true"
      class="c2 c3 c1"
      role="dialog"
    >
      <div
        aria-describedby="Modal-3"
        aria-labelledby="Modal-4"
        class="c4 c5 c1"
        role="dialog"
        style="width: 100%;"
        tabindex="-1"
        width="100%"
      >
        <div
          class="c6 c1"
        >
          <h5
            class="c7 sub-heading c8 c1"
            id="Modal-4"
          >
            This is a title
          </h5>
        </div>
        <div
          class="c9 c1"
          id="Modal-3"
        >
          This is the content
        </div>
        <div
          class="c10 c1"
          style="justify-content: flex-end;"
        >
          <div
            class="c1"
          >
            <div
              class="c11 c1"
              spacing="xxsmall"
            >
              <button
                class="c12 c13 "
                kind="default"
                style="size: default;"
                type="button"
              >
                Cancel
              </button>
              <button
                class="c14 c15"
                kind="default"
                style="size: default;"
                type="button"
              >
                Submit
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</portal>
`;

exports[`renders correctly for a dialog modal with type alert 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 998;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  -moz-tap-highlight-color: rgba(0,0,0,0);
  background-color: black;
  opacity: 0.4;
}

.c0[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

.c3 {
  -webkit-transform: translate3d(-50%,-50%,0px);
  -ms-transform: translate3d(-50%,-50%,0px);
  transform: translate3d(-50%,-50%,0px);
  position: fixed;
  z-index: 19900410;
  left: 50%;
  top: 50%;
}

.c3[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

.c2 {
  padding: 0 1rem;
  max-width: 600px;
  width: 100%;
}

.c5 {
  background-color: white;
  border-radius: 3px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border: 1px solid #e6e6e6;
}

.c6 {
  padding: 1.25rem;
  max-height: 50vh;
  overflow-y: scroll;
  width: 100%;
}

.c4 {
  border-radius: 5px;
}

.c4:focus {
  outline: unset;
}

<portal>
  <div>
    <div
      aria-hidden="false"
      class="c0 c1"
      role="button"
      tabindex="-1"
    />
    <div
      aria-hidden="false"
      aria-modal="true"
      class="c2 c3 c1"
      role="dialog"
    >
      <div
        aria-describedby="Modal-3"
        aria-labelledby="Modal-4"
        class="c4 c5 c1"
        role="alertdialog"
        style="width: 100%;"
        tabindex="-1"
        width="100%"
      >
        <div
          class="c6 c1"
          id="Modal-3"
        >
          This is the content
        </div>
      </div>
    </div>
  </div>
</portal>
`;
